packages/cyberstorm/src/components/Breadcrumb.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { Fragment } from "react";
+import styles from "./componentStyles/Breadcrumb.module.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faHouse, faSlashForward } from "@fortawesome/pro-regular-svg-icons";
+import { LinkingProvider } from "@thunderstore/components";
+import { LinkLibrary } from "./LinkLibrary";
+
+interface Crumb {
+  label: string;
+  LinkComponent?: (props: any) => React.ReactElement | null; // eslint-disable-line @typescript-eslint/no-explicit-any
+  LinkProps?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+interface Props {
+  parts: Crumb[];
+}
+
+/**
+ * Navigational bread crumbs defining where the user is at the site.
+ */
+export const BreadCrumb: React.FC<Props> = (props) => {
+  if (!props.parts) {
+    return (
+      <div className={`${styles.breadcrumb}`}>
+        <Home />
+      </div>
+    );
+  }
+  const parts = [...props.parts];
+  const leaf = parts.pop();
+
+  return (
+    <div className={`${styles.breadcrumb}`}>
+      <Home />
+      {(parts.length || leaf !== undefined) && <Separator />}
+
+      {parts.map((crumb, i) => (
+        <Fragment key={`${crumb.label}-${i}`}>
+          <MidCrumb crumb={crumb} />
+          <Separator />
+        </Fragment>
+      ))}
+
+      <LeafCrumb crumb={leaf} />
+    </div>
+  );
+};
+
+const Home: React.FC = () => {
+  return (
+    <LinkingProvider value={LinkLibrary}>
```

Comment:
Second issue: LinkingProvider should be used on the apps using the components, not in the component library. Ideally it's included just once in the top of the app ([[1]](https://github.com/thunderstore-io/thunderstore-ui/blob/master/apps/nextjs/pages/_app.tsx#L31), [[2]](https://github.com/thunderstore-io/thunderstore-ui/blob/master/apps/storybook/.storybook/preview.js#L27)), instead of wrapping each and every link.

Reply:
Changed it so that `BreadCrumb` needs a linkLibrary as a prop. Is that the correct way?

Reply:
Oh no hold on I'm not sure anymore. Do I need to pass the linkLibrary to the function or can I fetch somehow with the `useContext`?

Reply:
Methinks you should do just like we do in the existing example we have:

1. The [app defines a `LinkLibrary`](https://github.com/thunderstore-io/thunderstore-ui/blob/master/apps/storybook/LinkLibrary.tsx), telling how the link should be rendered in that app
2. Somewhere [high in the app structure, the app uses `LinkingProvider`](https://github.com/thunderstore-io/thunderstore-ui/blob/master/apps/storybook/.storybook/preview.js#L27), passing the `LinkLibrary` to it as a prop. All the wrapped components can now consume the context of that provider.
3. In the app, inside that context, a [component defines which links should be used](https://github.com/thunderstore-io/thunderstore-ui/blob/master/apps/storybook/stories/components/BreadCrumbs.stories.tsx#L26-L28) by the breadcrumbs component. The ["links" are in fact components themselves](https://github.com/thunderstore-io/thunderstore-ui/blob/master/apps/storybook/stories/components/BreadCrumbs.stories.tsx#L4-L5), defined in the component library
4. The [breadcrumbs component merely renders the link components](https://github.com/thunderstore-io/thunderstore-ui/blob/master/packages/components/src/components/BreadCrumbs.tsx#L63-L65) passed to it. It doesn't know what's the output when it renders them, and much like a honeybadger, it doesn't care. As long as the rendering happens within the context of `LinkingProvider`, the output will be what was defined in the `LinkLibrary`

Reply:
I think I've implemented it correctly and according to your guide. (which was a good one and maybe something we might want to put into the LinkingProvider readme)

Though I do have to note that I did not add this into any "app", since we don't have an app yet.

I've added the usage of `LinkingProvider` into the `Breadcrumb.stories.tsx`, as a decorator for the story.

Reply:
> Though I do have to note that I did not add this into any "app", since we don't have an app yet.

I would consider `cyberstorm-storybook`, which resides under the `apps` dir, an "app" 😉